examples/xds-management-server/cmd/controller.go

```diff
@@ -0,0 +1,158 @@
+ * limitations under the License.
+ */
+
+package main
+
+import (
+	"context"
+	"os"
+	"os/signal"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"quilkin.dev/xds-management-server/pkg/cluster"
+	"quilkin.dev/xds-management-server/pkg/filterchain"
+
+	agones "agones.dev/agones/pkg/client/clientset/versioned"
+
+	"k8s.io/apimachinery/pkg/util/clock"
+
+	"agones.dev/agones/pkg/client/informers/externalversions"
+	agonescluster "quilkin.dev/xds-management-server/pkg/cluster/agones"
+	k8sfilterchain "quilkin.dev/xds-management-server/pkg/filterchain/k8s"
+	"quilkin.dev/xds-management-server/pkg/k8s"
+
+	"github.com/alecthomas/kong"
+	log "github.com/sirupsen/logrus"
+
+	"quilkin.dev/xds-management-server/pkg/server"
+	"quilkin.dev/xds-management-server/pkg/snapshot"
+)
+
+type flags struct {
+	Port                    int16         `name:"port" help:"Server listening port." default:"18000"`
+	ProxyNamespace          string        `name:"proxy-namespace" help:"Namespace under which the proxies run." default:"quilkin"`
+	GameServersNamespace    string        `name:"game-server-namespace" help:"Namespace under which the game-servers run." default:"gameservers"`
+	GameServersPollInterval time.Duration `name:"game-server-poll-interval" help:"How long to wait in-between checking for game-server updates." default:"1s"`
+	ProxyPollInterval       time.Duration `name:"proxy-interval" help:"How long to wait in-between checking for proxy updates." default:"1s"`
+}
+
+func createAgonesClusterProvider(
+	ctx context.Context,
+	logger *log.Logger,
+	k8sConfig *rest.Config,
+	flags *flags,
+) cluster.Provider {
+	agonesClient, err := agones.NewForConfig(k8sConfig)
+	if err != nil {
+		log.WithError(err).Fatal("failed to create Agones clientset")
+	}
+
+	informerFactory := externalversions.NewSharedInformerFactoryWithOptions(
+		agonesClient,
+		0,
+		externalversions.WithNamespace(flags.GameServersNamespace))
+	informerFactory.Start(ctx.Done())
+	gameServerInformer := informerFactory.Agones().V1().GameServers().Informer()
+
+	return agonescluster.NewProvider(logger, gameServerInformer, agonescluster.Config{
+		GameServersNamespace:    flags.GameServersNamespace,
+		GameServersPollInterval: flags.GameServersPollInterval,
+	})
+}
+
+func createFilterChainProvider(
+	ctx context.Context,
+	logger *log.Logger,
+	k8sClient *kubernetes.Clientset,
+	flags *flags,
+) filterchain.Provider {
+
+	informerFactory := informers.NewSharedInformerFactoryWithOptions(
+		k8sClient,
+		0,
+		informers.WithNamespace(flags.ProxyNamespace),
+		informers.WithTweakListOptions(func(options *metav1.ListOptions) {
+			options.LabelSelector = k8sfilterchain.LabelSelectorProxyRole
+		}),
+	)
+	informerFactory.Start(ctx.Done())
+
+	return k8sfilterchain.NewProvider(
+		logger,
+		clock.RealClock{},
+		informerFactory.Core().V1().Pods().Lister(),
```

Comment:
So you use the Lister here, but not above 😄 Would be good to be consistent between the two.